code/modules/mob/living/carbon/human/human.dm

```diff
@@ -240,22 +240,24 @@
 				limbs_amount = 4
 
 		if(EXPLODE_HEAVY)
+			var/stuntime = 5 SECONDS - (bomb_armor / 20) SECONDS //Between no stun and 5 seconds of stun depending on bomb armor
 			brute_loss = 60
 			burn_loss = 60
 			if(bomb_armor)
 				brute_loss = 30 * (2 - round(bomb_armor * 0.01, 0.05))
 				burn_loss = brute_loss				//Damage gets reduced from 120 to up to 60 combined brute+burn
 			if(check_ear_prot() < HEARING_PROTECTION_TOTAL)
 				AdjustEarDamage(30, 120)
-			Weaken(20 SECONDS - (bomb_armor * 1.6 / 10) SECONDS) 	//Between ~4 and ~20 seconds of knockdown depending on bomb armor
+			Weaken(stuntime)
+			KnockDown(stuntime * 3) //Up to 15 seconds of knockdown
 
 		if(EXPLODE_LIGHT)
 			brute_loss = 30
 			if(bomb_armor)
 				brute_loss = 15 * (2 - round(bomb_armor * 0.01, 0.05)) //Reduced from 30 to up to 15
 			if(check_ear_prot() < HEARING_PROTECTION_TOTAL)
 				AdjustEarDamage(15, 60)
-			Weaken(16 SECONDS - (bomb_armor * 1.6 / 10) SECONDS) //Between no knockdown to ~16 seconds depending on bomb armor
+			KnockDown(10 SECONDS - (bomb_armor / 10) SECONDS) //Between no knockdown to 10 seconds of knockdown depending on bomb armor
```

Comment:
```suggestion
			KnockDown(10 SECONDS - bomb_armor) //Between no knockdown to 10 seconds of knockdown depending on bomb armor
```
same thing here. might want to mention that bomb armour is 0-100? I dunno. also this stuff will need to be changed in #18547

Reply:
I am so fucking stupid [2]